tools/alerts/alert_registry.py

```diff
@@ -0,0 +1,119 @@
+CANCELED = "canceled"
+
+# rename this when these are ready
+# PYTORCH_ALERT_LABEL = "pytorch-alert"
+PYTORCH_ALERT_LABEL = "pytorch-alert-test"
+
+headers = {"Authorization": f"token {os.environ.get('GITHUB_TOKEN')}"}
+
+def register_alert(alert_type):
+    if alert_type in ALERT_REGISTRY:
+        raise ValueError(f"Alert type {alert_type} is already registered")
+    def inner(func):
+        ALERT_REGISTRY[alert_type] = func
+        return func
+    return inner
+
+def _assert_same_repo_and_type(alerts: List[Dict[str, Any]]) -> None:
+    repo = alerts[0]["repo"]
+    alert_type = alerts[0]["AlertType"]
+    alert_org = alerts[0]["organization"]
+    for alert in alerts:
+        if alert["repo"] != repo:
+            raise ValueError(
+                f"Alerts must be from the same repository, got {repo} and {alert['repo']}"
+            )
+        if alert["AlertType"] != alert_type:
+            raise ValueError(
+                f"Alerts must be of the same type, got {alert_type} and {alert['AlertType']}"
+            )
+        if alert["organization"] != alert_org:
+            raise ValueError(
+                f"Alerts must be of the same org, got {alert_org} and {alert['org']}"
+            )
+
+def _assert_same_branch(alerts: List[Dict[str, Any]]) -> None:
+    branch = alerts[0]["branch"]
+    for alert in alerts:
+        if alert["branch"] != branch:
+            raise ValueError(
+                f"Alerts must be from the same branch, got {branch} and {alert['branch']}")
+
+@register_alert('Recurrently Failing Job')
+def handle_recurrently_failing_jobs(alerts: List[Dict[str, Any]]) -> Any:
+    _assert_same_repo_and_type(alerts)
+    _assert_same_branch(alerts)
+    repo = alerts[0]["repo"]
+    issue =  generate_failed_job_issue(alerts)
+    existing_alerts = fetch_alerts_filter(repo, [PYTORCH_ALERT_LABEL], alerts[0]["AlertType"])
+    clear_alerts(existing_alerts[:-1])
+    if len(existing_alerts) == 0:
+        existing_alerts.push(issue)
+        create_issue(issue)
+    else:
+        update_issue(issue, existing_alerts[-1])
+    return issue
+
+def generate_failed_job_hud_link(failed_job_name: str) -> str:
+    # TODO: I don't think minihud is universal across multiple repositories
+    #       would be good to just replace this with something that is
+    hud_link = "https://hud.pytorch.org/minihud?name_filter=" + urllib.parse.quote(
+        failed_job_name
+    )
+    return f"[{failed_job_name}]({hud_link})"
+
+def generate_failed_job_issue(
+    alerts
+) -> Any:
+    alerts.sort(key=lambda alert: alert["timestamp"])
+    alerts.reverse()
+    issue = {}
+    issue["state"] = "closed"
+    issue[
+        "title"
+    ] = f"[{alerts[0]['repo']}] [{alerts[0]['AlertType']}] [Branch: {alerts[0]['branch']}] [TEST ALERT PAY NO ATTENTION TO THIS] There are {len([alert for alert in alerts if not alert['closed']])} failing jobs"
+    body = "There are the following failures on the main branch of pytorch: \n"
```

Comment:
I think the branch name here needs to come from `alerts[0]['branch']`

Reply:
wait it does right?

Reply:
Oh I mean the body message say `There are the following failures on the main branch of pytorch` but I think it should refer to the repo the branch in the alerts, similar to `issue["title"]`, shouldn't it?